### PR TITLE
feat: on_demand flows, attrbute & rental changes

### PR DESCRIPTION
**Pull request by anuj-ondc**

### CODE REVIEW

File: api/components/enum/cancellation.yaml [hunk 1]
```diff
@@ -0,0 +1,19 @@
+descriptor:
```
Comment (92shreyansh) [inline api/components/enum/cancellation.yaml]: These should be same as fulfilment.state.descriptor.code

File: api/components/examples/on-demand/cancel/cancel_0.yaml [hunk 2]
```diff
@@ -16,6 +16,7 @@ value:
         code: IND
       city:
         code: std:080
+    ttl: PT30S    
   message:
     cancellation_reason_id: "5"
-    order_id: b4232ad4-19ee-4c67-9223-a5189b13a741
+    order_id: "01"
```
Comment (92shreyansh) [inline api/components/examples/on-demand/cancel/cancel_0.yaml]: this is O1 not 01

File: api/components/examples/on-demand/on_init/on_init_1.yaml [hunk 3]
```diff
@@ -0,0 +1,177 @@
...
+            - type: START
+              location:
+                gps: 13.008935, 77.644408
+              time:
+                timestamp: "2023-08-23T13:00:40.065Z"
+            - type: END
+              location:
+                gps: 12.971186, 77.586812
+          vehicle:
+            category: AUTO_RICKSHAW
```
Comment (92shreyansh) [inline api/components/examples/on-demand/on_init/on_init_1.yaml:110]: lets change the catagory to CAB for rental and schedule

Comment (anuj-ondc) [reply api/components/examples/on-demand/on_init/on_init_1.yaml:110]: updated

File: api/components/examples/on-demand/cancel/cancel_0.yaml (hunk 2 above)
Comment (anuj-ondc) [reply api/components/examples/on-demand/cancel/cancel_0.yaml]: updated

File: api/components/enum/cancellation.yaml (hunk 1 above)
Comment (anuj-ondc) [reply api/components/enum/cancellation.yaml]: updated

**Pull request merged**